gateway/src/api/latest.rs

```diff
@@ -103,161 +93,6 @@ impl StatusResponse {
-    let page = page.unwrap_or(0);
-    let projects = service
-        // The `offset` is page size * amount of pages
-        .iter_user_projects_detailed(&name, limit * page, limit)
-        .await?
-        .map(|project| project::Response {
-            name: project.0.to_string(),
-            state: project.1.into(),
-        })
-        .collect();
-
-    Ok(AxumJson(projects))
-}
-
-#[instrument(skip_all, fields(%project))]
-#[utoipa::path(
-    post,
-    path = "/projects/{project_name}",
-    request_body = shuttle_common::models::project::Config,
-    responses(
-        (status = 200, description = "Successfully created a specific project.", body = shuttle_common::models::project::Response),
-        (status = 500, description = "Server internal error.")
-    ),
-    params(
-        ("project_name" = String, Path, description = "The name of the project."),
-    )
-)]
-async fn create_project(
-    State(RouterState {
-        service, sender, ..
-    }): State<RouterState>,
-    User { name, claim, .. }: User,
-    Path(project): Path<ProjectName>,
-    AxumJson(config): AxumJson<project::Config>,
-) -> Result<AxumJson<project::Response>, Error> {
-    let is_admin = claim.scopes.contains(&Scope::Admin);
-
-    let state = service
-        .create_project(project.clone(), name.clone(), is_admin, config.idle_minutes)
-        .await?;
-
-    service
-        .new_task()
-        .project(project.clone())
-        .send(&sender)
-        .await?;
-
-    let response = project::Response {
-        name: project.to_string(),
-        state: state.into(),
-    };
-
-    Ok(AxumJson(response))
-}
-
-#[instrument(skip_all, fields(%project))]
-#[utoipa::path(
-    delete,
-    path = "/projects/{project_name}",
-    responses(
-        (status = 200, description = "Successfully destroyed a specific project.", body = shuttle_common::models::project::Response),
-        (status = 500, description = "Server internal error.")
-    ),
-    params(
-        ("project_name" = String, Path, description = "The name of the project."),
-    )
-)]
-async fn destroy_project(
-    State(RouterState {
-        service, sender, ..
-    }): State<RouterState>,
-    ScopedUser { scope: project, .. }: ScopedUser,
-) -> Result<AxumJson<project::Response>, Error> {
-    let state = service.find_project(&project).await?;
-
-    let mut response = project::Response {
-        name: project.to_string(),
-        state: state.into(),
-    };
-
-    if response.state == shuttle_common::models::project::State::Destroyed {
-        return Ok(AxumJson(response));
-    }
-
-    // if project exists and isn't `Destroyed`, send destroy task
-    service
-        .new_task()
-        .project(project)
-        .and_then(task::destroy())
-        .send(&sender)
-        .await?;
-
-    response.state = shuttle_common::models::project::State::Destroying;
-
-    Ok(AxumJson(response))
-}
-
-#[instrument(skip_all, fields(scope = %scoped_user.scope))]
-async fn route_project(
```

Comment:
I am not sure how you thought about it (maybe adding a similar handler in a new PR), but something similar to this will be needed.

Reply:
Yes, indeed, we will need to add that